ztp/resource-generator/entrypoints/generator

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env bash
+
+PROG=$(basename $0)
+SUB_COMMAND=$1
+
+# sub command-line arguments
+EMS_ONLY="false"
+WRAP_IN_POLICY="true"
+
+ztpTmpDir="/home/ztp/tmp"
+basePath="/kustomize/plugin/ran.openshift.io/v1"
+policyGenPath=${basePath}/policygentemplate
+siteConfigPath=${basePath}/siteconfig
+
+printenv MOUNT_DIR &> /dev/null
+if [[ $? -ne 0 ]]; then
+  # default dest dir in the container that is mounted from the host
+  MOUNT_DIR="/home/ztp_resources"
+fi
+
+function help {
+    cat >&2 <<EOF
+ztp-site-generator single site resources generator
+
+Usage: ${PROG} <subcommand> [options]
+
+Available commands:
+  installCRs          generate installation CRs of given SiteConfig(s)
+  configurationCRs    generate configuration CRs of given PolicyGenTemplate(s)
+
+Options:
+  --help Help for generator
+
+EOF
+}
+
+function installCRs_help {
+    cat >&2 <<EOF
+Generate the installation CRs based on SiteConfig CR
+
+Description:
+  This command takes the SiteConfig file(s) to generate the following CRs that can be used by Assisted Installer for cluster install:
+  * AgentClusterInstall
+  * ClusterDeployment
+  * NMStateConfig
+  * KlusterletAddonConfig
+  * ManagedCluster
+  * InfraEnv
+  * BareMetalHost
+  * HostFirmwareSettings
+  * extra-manifest configurations
+
+Environment variables:
+    \$MOUNT_DIR          set the dest path of the volume mounted in from the host (default "/home/ztp_resources"). It's the base location for storing the input and output resources.
+
+The SRC_PATH and DEST_DIR are paths in the container and they should be container volumes mounted from the host for you to access the generated resources on the host.
+Both SRC_PATH and DEST_DIR can be relative path to MOUNT_DIR. The SRC_PATH can be a SiteConfig yaml or a directory contains SiteConfig instances.
+The DEST_DIR should be a directory and if DEST_DIR is not sepecified, by default the generated resources will be outputted to \$MOUNT_DIR/out/generated_installCRs.
+
+Usage: ${PROG} ${SUB_COMMAND} SRC_PATH [DEST_DIR] [options]
+
+Options:
+  -h, --help                           help
+  -e, --extra-manifest-only=false      If true, generate extra manifests only
+
+Examples:
+  # If you have SiteConfig(s) stored in /home/ztp_resources/siteconfigs on the host,
```

Comment:
I think if I had my SiteConfigs in a directory like that, I'd want to just mount that directly into the container and not have to specify a sub-directory.  For example:

```
podman run -v /home/ztp_resources/siteconfigs:/resources -v /home/ztp_resources/siteconfigs/outputs:/output generator siteconfig -e
```